src/Ingress.Asb.Worker/ServiceBusClient.cs

```diff
@@ -81,84 +81,93 @@ private async Task ProcessMessagesAsync(Message message, CancellationToken token
 
-                    var roadSegmentPatch = new RoadSegment(roadSegID, surfaceType, prediction.Probability);
+                        var settings = new JsonSerializerSettings
+                        {
+                            ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                            NullValueHandling = NullValueHandling.Ignore
+                        };
 
-                    var settings = new JsonSerializerSettings
-                    {
-                        ContractResolver = new CamelCasePropertyNamesContractResolver(),
-                        NullValueHandling = NullValueHandling.Ignore
-                    };
+                        var roadSegJson = JsonConvert.SerializeObject(roadSegmentPatch, settings);
+                        var data = new StringContent(roadSegJson, Encoding.UTF8, "application/ld+json");
 
-                    var roadSegJson = JsonConvert.SerializeObject(roadSegmentPatch, settings);
-                    var data = new StringContent(roadSegJson, Encoding.UTF8, "application/ld+json");
+                        var patchResponse = await client.PatchAsync(patchURL, data);
 
-                    var patchResponse = await client.PatchAsync(patchURL, data);
+                        if (patchResponse.IsSuccessStatusCode) {
+                            var stringContent = await patchResponse.Content.ReadAsStringAsync();
+                            _logger.LogInformation(patchResponse.StatusCode + ": " + stringContent);
+                        } else {
+                            _logger.LogError(
+                                "Failed to patch road segment {RoadSegment}: {StatusCode} {Content}",
+                                roadSegID, patchResponse.StatusCode, patchResponse.Content
+                                );
+                        }
 
-                    if (patchResponse.IsSuccessStatusCode) {
-                        var stringContent = await patchResponse.Content.ReadAsStringAsync();
-                        _logger.LogInformation(patchResponse.StatusCode + ": " + stringContent);
                     } else {
-                        _logger.LogError(
-                            "Failed to patch road segment {RoadSegment}: {StatusCode} {Content}",
-                            roadSegID, patchResponse.StatusCode, patchResponse.Content
+                        _logger.LogWarning(
+                            "No road segments found within {Distance}m of ({longitude}, {latitude}).",
+                            _maxRoadSegmentDistance, longitude, latitude
                             );
                     }
-
+                    
+                    // Complete the message so that it is not received again.
+                    // This can be done only if the subscriptionClient is created in ReceiveMode.PeekLock mode (which is the default).
+                    await _subscriptionClient.CompleteAsync(message.SystemProperties.LockToken);
```

Comment:
You must Complete messages even if they have status == 0 otherwise you will continue to get the same message until it dead letters. We don't want that.